From 2a0c34eea8f68811f6eee2ad32652f20182d322e Mon Sep 17 00:00:00 2001
From: Alan Jenkins <alan.christopher.jenkins@gmail.com>
Date: Tue, 22 Aug 2017 13:54:21 +0100
Subject: [PATCH] logind: method_schedule_shutdown() already rejects empty
 `type`

Don't test for an empty `type` afterwards.  This is not how you cancel
scheduled shutdowns - there's a separate method for that.

(cherry picked from commit f8169e62dfc9ea903bc880064a5d384cdc68db3e)
---
 src/login/logind-dbus.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/src/login/logind-dbus.c b/src/login/logind-dbus.c
index 242ad4fb2b..f0943e56e0 100644
--- a/src/login/logind-dbus.c
+++ b/src/login/logind-dbus.c
@@ -2051,12 +2051,9 @@ static int method_schedule_shutdown(sd_bus_message *message, void *userdata, sd_
         if (r < 0)
                 return r;
 
-        if (!isempty(type)) {
-                r = update_schedule_file(m);
-                if (r < 0)
-                        return r;
-        } else
-                (void) unlink("/run/systemd/shutdown/scheduled");
+        r = update_schedule_file(m);
+        if (r < 0)
+                return r;
 
         return sd_bus_reply_method_return(message, NULL);
 }
